src/simulator_renaissance.cpp

```diff
@@ -0,0 +1,365 @@
+             */
+            if (line.length() == 12) {
+               time = line;
+            }else if (line.length() > 12) {
+                string payload = line; /**<  string line is copied to string payload */
+                /** 
+                 * .find() returns the index of the first occurrence of the substring("[]")
+                 *  in the string (payload) from the start.
+                 *  If the substring is found it returns std::string::npos which means -1. The
+                 *  following if conditional block checks for it, and if the statement becomes true,
+                 *  then the "continue", statement executes, and the code handle returns to while block.
+                 *  Otherwise, the block tries to parse the string for the required pattern.
+                */
+                size_t mes_status = payload.find("[]");
+                if (mes_status != std::string::npos) {
+                    continue;
+                }else{
+                    std::fill_n(arr, 5, 0); 		  /**<  Resets the arr elements with 0 */
+                    counter = 0;                      /**<  counter is reset */
+                    size_t pos = payload.find(comma); /**<  Finds the first occurence of comma in payload */
+                    while (pos != std::string::npos)  /**<  Keep running until there is no comma */
+                    {
+                        arr[counter + 1] = pos;       /**<  Store the location of first comma at position 1 in arr */
+                        ++counter; 				      /**< Increment the counter by 1 */
+                        pos = payload.find(comma, pos + comma.size()); /** Look for the next occurrence of comma starting after the 
+                                                                        * first occurrence of the previous comma*/
+                    }
+                    /**
+                     * The following line finds the position of model (payload.find("model")) in
+                     * the payload, and that position is added to (6) include the length of model
+                     * and space. This becomes the parameter of (.substr()) which returns the substring
+                     * starting from this parameter position till the end-of-line.
+                     */
+                    string component_check = payload.substr(payload.find("model") + 6);
+                    counter = 0;				      /**<  counter is reset */
+                    /**
+                     * The following do-while loop works until the first 0 is encountered in arr
+                     * starting after the 0th element position*/
+                    do {
+                        val_pos = payload.find("{", arr[counter]); /**< Find the first "{" position starting from arr value position */
+                        string value = payload.substr(val_pos + 1, payload.find("}", /**< Returns the substring between {} */ 
```

Comment:
Lines < 80 characters:  you should follow coding conventions. please do it for other lines as well.

Reply:
done